core/src/main/java/io/grpc/internal/RetriableStream.java

```diff
@@ -32,12 +34,17 @@
 import java.util.HashSet;
 import java.util.List;
 import java.util.Set;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.concurrent.atomic.AtomicLongFieldUpdater;
 import javax.annotation.CheckReturnValue;
 import javax.annotation.Nullable;
 import javax.annotation.concurrent.GuardedBy;
 
 /** A logical {@link ClientStream} that is retriable. */
 abstract class RetriableStream<ReqT> implements ClientStream {
+  @SuppressWarnings("rawtypes") // rawtypes is fine here
+  private static final AtomicLongFieldUpdater<RetriableStream> perRpcBufferUsedUpdater =
```

Comment:
The problem is that ReqT is not available in static context.  If I cast, I need to cast twice. Are you OK with this:
```
  @SuppressWarnings("unchecked") // ReqT type is irrelevant here.
  private static final AtomicLongFieldUpdater<RetriableStream<?>> perRpcBufferUsedUpdater =
      (AtomicLongFieldUpdater<RetriableStream<?>>) (AtomicLongFieldUpdater<?>) 
          AtomicLongFieldUpdater.newUpdater(RetriableStream.class, "perRpcBufferUsed");
```
?

Reply:
You may try (although it may need the two casts as well):
```
  private static final AtomicLongFieldUpdater<RetriableStream<?>> perRpcBufferUsedUpdater =
        AtomicLongFieldUpdater.newUpdater((Class<RetriableStream<?>>) RetriableStream.class, "perRpcBufferUsed");
```

But yes. I'd be more happy with the icky casts on this one line than fighting lack-of-generics oddities.

Reply:
Yeah, it need two casts as well
```java
(Class<RetriableStream<?>>) (Class<?>) RetriableStream.class
```